Library/Homebrew/formula_installer.rb

```diff
@@ -625,9 +661,10 @@ def pour
       downloader.stage
     end
 
-    keg = Keg.new(formula.prefix)
-    keg.relocate_install_names Keg::PREFIX_PLACEHOLDER, HOMEBREW_PREFIX.to_s,
-      Keg::CELLAR_PLACEHOLDER, HOMEBREW_CELLAR.to_s, :keg_only => formula.keg_only?
+    if !pour_bottle? || formula.bottle.needs_relocation?
```

Comment:
Sorry but you are incorrect. It will relocate both binaries and text files. https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/keg_relocate.rb#L41-L58